getting_started/checkpoints.md

```diff
@@ -0,0 +1,113 @@
+
+1. **Unstacked Checkpoints**: Parameters for each layer are stored in separate groups (e.g., `layers_0`, `layers_1`, `layers_2`). This is the common format for models released publicly. (Figure 1, left).
+2. **Stacked Checkpoints**: Corresponding parameters from all layers are combined into a single, larger tensor (e.g., all attention weights are in one array). This is the format `jax.lax.scan` expects. (Figure 1, right).
+
+![Illustration of an unstacked checkpoint versus a stacked checkpoint.](checkpoints_explain.png)
+*Figure 1: A comparison of an unstacked checkpoint and a stacked checkpoint for a simple language model.*
+
+Their difference can also be represented in the following pytree structure:
+```
+# Stacked (aka scanned)
+"params" : {
+"mlp_0" : [num_layers, emb_d, ff_d]
+"mlp_1" : [num_layers, ff_d, emb_d]
+}
+
+# Unstacked
+"params" {
+{"layer_0":
+   "mlp_0" : [ emb_d, ff_d]
+    "mlp_1"" : [ff_d,emb_d]
+"layer_1"
+...
+}
+```
+
+The stacked format is highly efficient but has one key requirement: all layers within the `scan` operation must have identical configurations. For models with heterogeneous layers (where layer configurations differ), stacking is not possible, and only unstacked checkpoints can be used.
+
+### Takeaways
+
+To summarize the four checkpoint types:
+
+- **Unstacked Inference Checkpoint:** The standard format for publicly released models (e.g., on HuggingFace). Contains only model weights, stored layer-by-layer. Ideal for starting new training runs or for inference.
+- **Unstacked Training Checkpoint:** Contains weights and optimizer state, stored layer-by-layer. Used to resume training for models that do not use `jax.lax.scan` (e.g., models with non-identical layers).
+- **Stacked Inference Checkpoint:** Contains only model weights, but they are stacked for `scan`-compatible inference. Created by stripping the optimizer state from a stacked training checkpoint.
+- **Stacked Training Checkpoint:** The default format for saving and resuming training runs within MaxText. Contains both weights and optimizer state in a stacked format, optimized for `jax.lax.scan`.
+
+In MaxText, we treat **Unstacked Inference Checkpoints** as the standard format for *importing* a base model. For *saving and resuming* training, MaxText uses **Stacked Training Checkpoints** by default. Support for all four types is available to accommodate various models and use cases.
+
+---
+
+## Checkpoint Conversion
+
+MaxText provides conversion scripts to transform publicly available checkpoints (which are typically "Unstacked Inference Checkpoints") into the format required for MaxText training. These scripts handle the necessary renaming and reshaping of tensor keys.
+
+### Example: Converting a DeepSeek Checkpoint
```

Comment:
This is a one-off script, which is neither the old nor the new recommended way. Instead, you could point to Yixuan's new method.